File: R/cppClass/R/NumericVector.R
setClass("NumericVector", slots = c(ref = ))

setMethod("+", signature(e1 = "NumericVector", e2 = "NumericVector"), function(e1, e2)
{
    "inside +"
})

setGeneric("l2norm", function(x) standardGeneric("l2norm"))

setMethod("l2norm", signature(x = "NumericVector"), function(x)
{
    "l2 method"
})

File: R/cppClass/src/NumericVector.cpp
#include <math.h>

class NumericVector
{
    double* data;
    int n;
    double l2norm();
}


double NumericVector::l2norm()
{
    out = 0.0;
    for(int i = 0; i < n; i++)
    {
        out += data[i] * data[i];
    }
    return sqrt(out);
}
